server/src/main/java/org/elasticsearch/index/codec/tsdb/ES87TSDBDocValuesEncoder.java

```diff
@@ -232,20 +261,32 @@ void decodeOrdinals(DataInput in, long[] out, int bitsPerOrd) throws IOException
         assert out.length == ES87TSDBDocValuesFormat.NUMERIC_BLOCK_SIZE : out.length;
 
         long v1 = in.readVLong();
-        int header = (int) (v1 & 0b11L);
-        if (header == 0b00 || header == 0b10) {
-            // first bit is zero -> single run
+        int encoding = Long.numberOfTrailingZeros(~v1);
```

Comment:
nit: it could be slightly easier later on if we stripped the encoding from v1 here
```suggestion
        int encoding = Long.numberOfTrailingZeros(~v1);
        v1 >>>= encoding + 1;
```

Reply:
done